### PR TITLE
Added backup xcarchive action

### PR DESCRIPTION
**Pull request by dral3x**
Hello there,

I created a new action for zipping and exporting a xcarchive.
This can be used this way

```
desc "Deploy a new AppStore build"
lane :appstore do

  raise "Please specify BTSYNC_FOLDER inside `.env` file before run this lane again".red unless ENV['BTSYNC_FOLDER']

  cocoapods

  xcarchive(
    scheme: "Studio Prod",
    configuration: "ProdRelease",
    sdk: "iphoneos"
    )

  # Save archive to BTSync folder
  backup_xcarchive(
    destination: "#{ENV['BTSYNC_FOLDER']}/iOS Studio",
    zip: true
    )

  deliver :skip_deploy

end
```

### CODE REVIEW

File: lib/fastlane/actions/backup_xcarchive.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      end
+
+      def self.output
+        [
+          ['BACKUP_XCARCHIVE_FILE', 'Path to your saved xcarchive (compressed) file']
+        ]
+      end
+
+      def self.author
+        '[dral3x]'
```
Comment (KrauseFx) [inline lib/fastlane/actions/backup_xcarchive.rb]: Please change to `['dral3x']`

**Comment by KrauseFx** [comment]
This looks great, thanks @dral3x.

The tests are failing though:
```
     Failure/Error: action.is_supported?(nil) # this will raise an exception if the method is not implemented
     RuntimeError:
       Implementing `is_supported?` for all actions is mandatory. Please update Fastlane::Actions::BackupXcarchiveAction
     # ./lib/fastlane/action.rb:52:in `is_supported?'
     # ./spec/actions_list_spec.rb:25:in `block (5 levels) in <top (required)>'
```

Please implement it as:
```ruby
def self.is_supported?(platform)
  [:ios, :mac].include?platform
end
```
as it's only relevant for iOS and Mac.

Also, could you added one or two tests testing the basic functionality?

File: lib/fastlane/actions/backup_xcarchive.rb (hunk 1 above)
Comment (dral3x) [inline lib/fastlane/actions/backup_xcarchive.rb]: Done. Thanks

**Comment by dral3x** [comment]
My bad. I lost some new stuff from fastlane.
I will add `is_supported` and some tests. Thanks!

**Comment by dral3x** [comment]
Added a couple of tests

**Comment by KrauseFx** [comment]
Looks great, thank you so much @dral3x :+1:

**Pull request merged**